components/browser/menu/src/main/java/mozilla/components/browser/menu/BrowserMenu.kt

```diff
@@ -124,19 +128,29 @@ open class BrowserMenu internal constructor(
             menuPositioningData.inferredMenuPlacement is BrowserMenuPlacement.AnchoredToBottom.ManualAnchoring) {
 
             val collapsingMenuIndexLimit = adapter.visibleItems.indexOfFirst { it.isCollapsingMenuLimit }
+            val stickyItemHeight = menuList?.layoutManager
+                ?.getChildAt(adapter.visibleItems.indexOfLast { it.isSticky })
+                ?.height ?: 0
+
             if (collapsingMenuIndexLimit > 0) {
                 return ExpandableLayout.wrapContentInExpandableView(
                     view,
-                    collapsingMenuIndexLimit
+                    collapsingMenuIndexLimit,
+                    offset = stickyItemHeight
                 ) { dismiss() }
             }
-        }
-
-        // When showing an expandable bottom menu it should always be scrolled to the top (default in LayoutManager).
-        // Otherwise try showing the bottom of the menu when not enough space to fit it on the screen.
-        if (endOfMenuAlwaysVisible) {
-            menuList?.let {
-                showMenuBottom(it)
+        } else {
+            // The menu is by default set as a bottom one. Reconfigure it as a top one.
+            menuList?.layoutManager = StickyItemsLinearLayoutManager.get<BrowserMenuAdapter>(
+                view.context, StickyItemPlacement.TOP
+            )
+
+            // An expandable bottom menu should always be scrolled to the top (default in LayoutManager).
+            // Otherwise try showing the bottom of the menu when not enough space to fit it on the screen.
```

Comment:
If this `else` is for the top toolbar then these comments should be updated

Reply:
I think that is a correct description.
By default the menu is scrolled to the top. And this is what we want for the bottom menu. While for the top menu can be configured to start scrolled to the bottom.
Open to suggestion about what should say.

Reply:
If something is unclear it's better to fix it while we're at it. Reworded that to
> // By default the menu is laid out from and scrolled from top.
> // For the top menu it may be desired to initially show the bottom most items.
Hope this is better.